guides/source/getting_started.md

```diff
@@ -1991,6 +1991,191 @@ the `app/views/comments` directory.
 The `@article` object is available to any partials rendered in the view because
 we defined it as an instance variable.
 
+
+### Using Concerns
+
+A concern is any module that contains methods representing a well-defined slice of the functionality that a model or controller is responsible for. In other frameworks they are often known as mixins. Concerns are a way to make large controllers or models easier to understand and manage. This also has the advantage of reusability when multiple models (or controllers) share the same concerns.
```

Comment:
While accurate, I'd hesitate to describe a concern as a module.

I think concerns in Rails are primarily a design approach / style / pattern to break down models into separate (and reusable) slices of responsibility. Usage of modules is an implementation detail. 

Here's a [different approach](http://andrzejonsoftware.blogspot.com/2011/02/dci-and-rails.html) which is also implemented with modules containing well defined slices of functionality

```
class Article
end

module Notifier
  def send_notifications
    ...
  end
end

class ArticlesController
  def create
    @article.extends(Notifier)

    if @article.save
       @article.send_notifications
    end
  end
end

Reply:
Perhaps reusing "a way to" makes it clearer that it's a pattern instead of tying it to modules?

```suggestion
A concern is a way to represent a well-defined slice of the functionality that a model or controller is responsible for. In other frameworks they are often known as mixins. They help make large controllers or models easier to understand and manage. This also has the advantage of reusability when multiple models (or controllers) share the same concerns.
```

Reply:
> Perhaps reusing "a way to" makes it clearer that it's a pattern instead of tying it to modules?

I like the flow, think the mixin line is more of a reference to modules though.

Reply:
@fylooi @sunny I just made a new revision - Swapped around the order of the sentences. I'm not sure if the mixin line is still worth including or not as a definition of modules might not be needed in a Rails guide, but I've kept it so far because it seems like a useful comparison to make to quickly build up understanding for those that are new to Rails.

Reply:
Thank you! 👏 

> I'm not sure if the mixin line is still worth including

I'd be inclined to remove it. To me, mixins are how we call "includable modules" in Ruby. I'd say that concerns are mixins so I feel that "In other languages, modules are often known as mixins" doesn't quite fit.

Reply:
And that is the fundamental issue with documenting "concerns", concerns are modules, there's not a lot to say about them. You are a Ruby programmer, you know what modules are and their use cases. If you don't, you'll find in Ruby learning resources. Rails guides should not teach threads in Ruby, should not teach method dispatch in Ruby, and should not teach code reuse via mixins in Ruby.

Reply:
In Ruby you split code in modules. If some module is reusable by several models, you put it in `app/models/concerns`. That's it.

On the other hand, Active Support has something called `ActiveSupport::Concern` that you do not need at all for writing concerns. A concern is a module. `AS::Concern` has some sugar for certain module inclusion patterns, and would have its own module documentation in the API.

And that is all.

Reply:
@fxn When I [first created this PR](https://github.com/rails/rails/pull/39766/commits/9aff8252f2e1dd65f52768c4da6719a7ffb9e480), given your previous feedback on the old PR, I actually had written the introduction as basically what you have here - concerns are modules, they help you reuse code. Does that seem like a better version than what is currently in the PR?

Aside from that, I think there's value in demonstrating how to use `ActiveSupport::Concern` or at least linking to the API docs from this guide. There's precedence for demonstrating the use of ActiveRecord or ActiveSupport in these guides - [here](https://guides.rubyonrails.org/active_record_basics.html) and [here](https://guides.rubyonrails.org/active_support_core_extensions.html) -- even though [they already have API guides](https://api.rubyonrails.org/classes/ActiveRecord/Base.html) - and from the feedback I'm seeing, there's value in having them as part of these guides as opposed to the API docs alone, which doesn't explicitly state the use for the folders. 

For me personally, what it comes down to is that a basic Rails app contains empty folders within prominent directories (`models` and `controllers`) that are not explicitly explained. If the folders were called `modules` instead of `concerns`, that would be pretty self-explanatory (even though I would personally still wonder why they are included if they are empty, when each user can create modules in any directory structure they like). Their presence and the name `concerns` implies a Rails convention. So as long as the folders exist with a name that is Rails-specific, we should have documentation somewhere intuitive to explain them - whether that's a tutorial like what I've written here, or just a paragraph or two (which the [previous PR was reduced to](https://github.com/rails/rails/pull/21114).) I totally agree with you that the Rails guides should not be responsible for demonstrating underlying Ruby concepts - but from what I can tell given the feedback I'm seeing, many people do not think `concerns` are just modules, and therefore they are not an underlying Ruby concept. 

I've always considered the Rails guides to be where you go to get wider context around how, when, and _why_ to use various Rails methods and conventions, whereas each page of the API guide is much more limited in scope. That is why I think the Rails guides should have some documentation for concerns (or remove the folders from Rails entirely, so we no longer imply a Rails convention if there isn't one). What the scope should be - I think that's the main challenge now. I can go either way - I think there is value in explaining the conceptual methodology, but if we can get a consensus on focusing these docs on modules without limiting the definition of concerns (as in @mcgregordan 's feedback) , I'm happy to go with that too. :) 

Tagging @sunny and @fylooi also - sorry this became a bit of an essay!

Reply:
> I've always considered the Rails guides to be where you go to get wider context around how, when, and why to use various Rails methods and conventions

This is also exactly what I think of the Rails guides at and the proposed documentation changes in this PR already function very well for this purpose - it's easy to see how the examples given would extend to other domains and your own problems you're using rails to solve.

I just don't want to see the added value of what's there get overlooked in the chase for something that's perhaps even better, especially if that version involves explaining less than the current version does.

Reply:
> On the other hand, Active Support has something called `ActiveSupport::Concern` that you do not need at all for writing concerns. A concern is a module. `AS::Concern` has some sugar for certain module inclusion patterns, and would have its own module documentation in the API.
> 
> And that is all.

While `ActiveSupport::Concern` can be used on its own, combining this sugar with the declarative DSL of an ActiveRecord model feels like a powerful and distinctly Rails convention to me. Think this particular pattern is worth highlighting.

Reply:
@kaspth @rafaelfranca what do you think of this PR so far? It would be great to hear if you share @fxn 's concerns around documentation or not - I made this PR based on feedback from others, but also your comments in my previous PR https://github.com/rails/rails/pull/21114 - so would be great to hear what you think! I haven't got any answer from a core team member since I responded.